### PR TITLE
chore: improve `cargo_clone_core` API

### LINKED ISSUE #42: Cargo clone as a library
Can I use cargo clone from my Rust app as a library rather than an executable?

It would be nice in my case, rather than spawning a process to run cargo clone.

### PR DESCRIPTION
**Pull request by MarcoIeni**
Closes #42

### CODE REVIEW

**Comment by JanLikar** [comment]
This looks nice!

File: cargo-clone-core/src/lib.rs [hunk 1]
```diff
@@ -29,72 +39,81 @@ pub use cargo::{
...
-    let should_append_crate_dir = multiple_crates || can_clone_in_dir;
+        let mut src = get_source(&self.srcid, &self.config)?;
 
-    // If prefix was not supplied, use current dir.
-    let dir_path = opts
-        .directory
-        .map(PathBuf::from)
-        .unwrap_or(env::current_dir()?);
+        self.clone_in(crate_, &self.directory, &mut src)
+    }
```
Comment (MarcoIeni) [inline cargo-clone-core/src/lib.rs:86]: when I used cargo-clone as a library, I found confusing that he was creating or not creating the directory, based on the number of crates I passed as arguments.
So I created two separate methods.

**Comment by MarcoIeni** [comment]
Ready for review! :)

File: cargo-clone/src/main.rs [hunk 2]
```diff
@@ -77,9 +78,9 @@ fn main() {
         .arg(Arg::with_name("directory").help("The destination directory. If it ends in a slash, crates will be placed into its subdirectories.").last(true));
 
     let matches = app.get_matches();
-    let mut config = Config::default().expect("Unable to get config.");
 
-    if let Err(e) = execute(matches, &mut config) {
+    if let Err(e) = execute(&matches) {
+        let config = cargo_config(&matches).expect("Unable to get config.");
```
Comment (MarcoIeni) [inline cargo-clone/src/main.rs:83]: Here I create a new cargo_config. Before, you were reusing the existing one.
Is it a problem?

Comment (JanLikar) [reply cargo-clone/src/main.rs:83]: I think this is fine.

**Review by JanLikar — APPROVED** [review]

**Review by JanLikar — CHANGES_REQUESTED** [review]
Hmm, looks like there's an error in some cases, let me test it out and get back to you.

**Comment by JanLikar** [comment]
Cloning without a directory failed eg. `cargo clone semver`.

I believe the code is correct now, but please double-check.

**Comment by MarcoIeni** [comment]
Oops, sorry!
Yeah, it seems good now 👍 
It's nice that we added another test case 😄

File: cargo-clone/tests/test_cli.rs [hunk 3]
```diff
@@ -28,14 +27,34 @@ fn test_cli() {
...
 }
 
+#[test]
+fn test_cli_no_directory() {
+    let temp_dir = TempDir::new("cargo-clone-tests").unwrap();
+    let output_path = temp_dir.path().join("cargo-clone");
+
+    assert!(!output_path.exists());
+
+    //assert!(temp_dir.path().exists());
```
Comment (MarcoIeni) [inline cargo-clone/tests/test_cli.rs:37]:
```suggestion
```

**Comment by JanLikar** [comment]
>  It's nice that we added another test case smile
Test-driven development FTW!

Thank you for the PR, I appreciate it.

**Review by JanLikar — APPROVED** [review]

**Pull request merged**